pkgs/misc/vim-plugins/update.py

```diff
@@ -111,6 +118,22 @@ def latest_commit(self) -> Tuple[str, datetime]:
             updated = datetime.strptime(updated_tag.text, "%Y-%m-%dT%H:%M:%SZ")
             return Path(str(url.path)).name, updated
 
+    def check_for_redirect(self, url: str, req: http.client.HTTPResponse):
+        response_url = req.geturl()
+        if url != response_url:
+            new_owner, new_name = (
+                urllib.parse.urlsplit(response_url).path.strip("/").split("/")[:2]
+            )
+            end_line = f" as {self.alias}\n" if self.alias else "\n"
+            plugin_line = "{owner}/{name}" + end_line
+
+            old_plugin = plugin_line.format(owner=self.owner, name=self.name)
```

Comment:
This seems very brittle to me, can't we use the original line instead? This breaks on any whitespace change.

Reply:
Or alternatively we could just rewrite the whole file and not bother replacing anything.

Reply:
I'm not sure what you mean by a "whitespace change". The `vim-plugin-names` file is rewritten in the same run of `update.py` as the redirections are detected here so why would the file change?

Reply:
I mean that this would break as soon as we'd have a line
```
owner/repo as  alias
```

(note the double space). The simplified solution would be to just remember the new lines (regardless of whether there was a redirect or not) and completely re-create the file instead of updating it. That would also simplify the sorting.

Reply:
I pushed a commit to normalize whitespace to deal with this particular issue. I recognize that this is still a bit flimsy but I would think it acceptable for a helper script, the results of which will be manually reviewed.

Reply:
> I pushed a commit to normalize whitespace to deal with this particular issue.

Why didn't you go for the complete rewrite? Seems more robust, easier and you get a sorted result "for free".

> I recognize that this is still a bit flimsy but I would think it acceptable for a helper script, the results of which will be manually reviewed.

I'd be careful with the "manual review" statement. These big automated PRs tend to be skimmed for malicious intent, but details tend to be skipped. At least that's how I do it. They are simply too big and too repetitive for detailed review.

Reply:
The file is rewritten every time now but I'm not sure if it's what you had in mind. It's not any more robust with respect to parsing the existing file.

Reply:
Its not *quite* what I had in mind. Now that we rewrite the file every time, there's not really a need to take the old file into consideration at all in `rewrite_input`. We could just re-generate an `plugin-names` file from scratch / from the data that was originally parsed. There would be no need to replace individual lines and worry about whitespace.

That's just a nitpick though, no reason to hold this off any longer.